lib/internal/util.js

```diff
@@ -187,3 +191,65 @@ exports.convertToValidSignal = function convertToValidSignal(signal) {
 
   throw new Error('Unknown signal: ' + signal);
 };
+
+const kCustomPromisifiedSymbol = Symbol('util.promisify.custom');
+const kCustomPromisifyArgsSymbol = Symbol('customPromisifyArgs');
+
+function promisify(orig) {
+  if (typeof orig !== 'function') {
+    throw new TypeError('The first argument to promisify() must be a function');
+  }
+
+  if (orig[kCustomPromisifiedSymbol]) {
+    const fn = orig[kCustomPromisifiedSymbol];
+    Object.defineProperty(fn, kCustomPromisifiedSymbol, {
+      value: fn, enumerable: false, writable: false, configurable: true
+    });
+    return fn;
+  }
+
+  if (isAsyncFunction(orig)) {
```

Comment:
@benjamingr hmm, I’ll think of it for a bit. I see the point in your example, but it doesn’t feel quite right to pass an extra `cb` in that case…

(Also, `util.promisify` would be exactly for that case, to enable users to avoid using the callback API ;))